weather_aplication/ShowWeatherViewController.swift

```diff
@@ -0,0 +1,193 @@
+import SnapKit
+import UIKit
+
+struct WeatherData {
+    let hour: String
+    let temperature: Double
+    let pressure: Double
+    let windSpeed: Double
+    let weatherCode: Int
+}
+
+class ShowWeatherViewController: UIViewController{
+    private let kSideMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    private let kBottomMargin: CGFloat = 5
+    private let kElementsToShow: Int = 24
+    private let kHeaderHeight: CGFloat = 50
+    
+    private let longitude: Double
+    private let latitude: Double
+    
+    private let cityLabel = UILabel()
+    private let weatherTableView = UITableView()
+    
+    private var date = String()
+    
+    private var dataArray: [WeatherData] = []
+    private var urlString: String {
+       let url =  "https://api.open-meteo.com/v1/forecast?latitude=\(latitude)&longitude=\(longitude)&hourly=temperature_2m,surface_pressure,weathercode,windspeed_10m"
+        return url
+    }
+    
+    func makeURLRequest() {
+        guard let queryURL =  URL(string:urlString) else { return }
+        let session = URLSession.shared
+        
+        session.dataTask(with: queryURL, completionHandler: { [weak self] data, response, error -> Void in
+            
+            if (error != nil) {
+                let alert = UIAlertController(title: "Error", message: error?.localizedDescription, preferredStyle: .alert)
+                alert.addAction(UIAlertAction(title: "OK", style: .default))
+                self?.present(alert, animated: true)
+            }
+            
+            if let httpResponse = response as? HTTPURLResponse {
+                if httpResponse.statusCode == 200 {
+                    do {
+                        guard let data = data else { return }
+                        let jsonResult = try JSONDecoder().decode(OpenMeteoJSON.self, from: data)
+                        self?.takeDataFromJson(jsonResult: jsonResult)
+                        DispatchQueue.main.async {
+                            self?.weatherTableView.reloadData()
+                        }
+                    } catch let error {
+                        print(error)
+                    }
+                }
+            }
+            
+        }).resume()
+    }
+    
+    func takeDataFromJson(jsonResult: OpenMeteoJSON) {
+        let elementsCount = kElementsToShow
+        let kCharsToDrop = 11
+        var dataArray2: [WeatherData] = []
+        if jsonResult.hourly.time.count >= kElementsToShow && jsonResult.hourly.temperature2M.count >= kElementsToShow && jsonResult.hourly.surfacePressure.count >= kElementsToShow && jsonResult.hourly.windspeed10M.count >= kElementsToShow && jsonResult.hourly.weathercode.count >= kElementsToShow {
```

Comment:
Jak mamy takiego ifa z dużą ilością warunków, to kolejne warunki możesz dać w nowej linijce. Też operator && możesz zasąpić po prostu przecinkiem